runtime/ts/entity.ts

```diff
@@ -51,11 +59,14 @@ export class Entity {
     }
     this[Symbols.identifier] = id;
   }
+
   toLiteral() {
     return this.rawData;
   }
 
-  static get type() {
+  abstract get key(): {tag: string, schema: Schema};
+
+  get type() {
```

Comment:
This needs to remain static.

Reply:
static get type() cannot be static because it references `this`.

Reply:
for static methods, 'this' refers to the class, not the instance.

Reply:
For static methods, `this` refers to the class rather than the instance.